pkg/ibm_plugin/sdk/vpn.go

```diff
@@ -0,0 +1,596 @@
+		config.SetDhGroup(24)
+		config.SetEncryptionAlgorithm(vpcv1.CreateIkePolicyOptionsEncryptionAlgorithmAes256Const)
+		config.SetKeyLifetime(27000)
+	}
+
+	ikePolicy, response, err := c.vpcService.CreateIkePolicy(config)
+	fmt.Print(response)
+	if err != nil {
+		return nil, err
+	}
+
+	return ikePolicy.ID, nil
+}
+
+// returns existing IKE policy for the peer cloud
+func (c *CloudClient) getIKEPolicy(peerCloud string) (*vpcv1.IkePolicy, error) {
+	ikePolicies, _, err := c.vpcService.ListIkePolicies(&vpcv1.ListIkePoliciesOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, policy := range ikePolicies.IkePolicies {
+		if strings.Contains(*policy.Name, peerCloud) {
+			return &policy, nil
+		}
+	}
+	// no policy matching the specified cloud was found
+	return nil, nil
+}
+
+// returns an IPsec policy ID that's compatible with the specified peer cloud
+func (c *CloudClient) getOrCreateIPSecPolicy(peerCloud string) (*string, error) {
+	// return an existing IPSec policy for the specified cloud if one exists in the region
+	existingPolicy, err := c.getIPSecPolicy(peerCloud)
+	if err != nil {
+		return nil, err
+	}
+	if existingPolicy != nil {
+		utils.Log.Printf("Using existing IPSec policy %v in region %v", *existingPolicy.Name, c.region)
+		return existingPolicy.ID, nil
+	}
+
+	// create a new IPSec policy for specified cloud
+	config := &vpcv1.CreateIpsecPolicyOptions{
+		Name:          core.StringPtr(GenerateResourceName("ipsec-" + peerCloud)),
+		ResourceGroup: c.resourceGroup,
+	}
+	if peerCloud == utils.AZURE {
+		config.SetAuthenticationAlgorithm("sha256")
+		config.SetEncryptionAlgorithm("aes256")
+		config.SetKeyLifetime(27000)
+		config.SetPfs(vpcv1.CreateIpsecPolicyOptionsPfsDisabledConst) // disable perfect forward secrecy
+	}
+	ipsecPolicy, _, err := c.vpcService.CreateIpsecPolicy(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return ipsecPolicy.ID, nil
+}
+
+// returns existing IPSec policy for the peer cloud
+func (c *CloudClient) getIPSecPolicy(peerCloud string) (*vpcv1.IPsecPolicy, error) {
+	ipSecPolicies, _, err := c.vpcService.ListIpsecPolicies(&vpcv1.ListIpsecPoliciesOptions{})
+	if err != nil {
+		return nil, err
+	}
+	for _, policy := range ipSecPolicies.IpsecPolicies {
+		if strings.Contains(*policy.Name, peerCloud) {
+			return &policy, nil
+		}
+	}
+	// no policy matching the specified cloud was found
+	return nil, nil
+}
+
+// returns the first available priority that doesn't cause a routing conflict.
+// if rule exists, returns false.
+// - routing conflict - occurs when 2 rules share the same zone, destination and priority.
+// - rule duplication - if specified rule matches a rule on the following fields: destination, zone and nextHopConnection.
+func (c *CloudClient) getAvailablePriority(routeData *vpcv1.CreateVPCRoutingTableRouteOptions) (bool, int64, error) {
+
+	var doesRuleExist bool
+	const numOfPriorities = 5
```

Comment:
Why 5?

Reply:
Priority values for VPC routing table routes are in the range: [0,4].